airflow/operators/sensors.py

```diff
@@ -222,6 +222,60 @@ def poke(self, context):
         return count
 
 
+class NamedHivePartitionSensor(BaseSensorOperator):
+    """
+    Waits for a partition to show up in Hive. Only accepts plain partition
+    names (eg dt=xxx) for the partition argument and no filters.
+
+    :param table: The name of the table to wait for, supports the dot
+        notation (my_database.my_table)
+    :type table: string
+    :param partitions: List of fully qualified names of the partitions to
+        wait for. This is passed as is to the Metastore Thrift client
+        "get_partitions_by_name" method.
+    :type partitions: list<string>
+    :param metastore_conn_id: reference to the metastore thrift service
+        connection id
+    :type metastore_conn_id: str
+    """
+
+    template_fields = ('partition_names', )
+
+    @apply_defaults
+    def __init__(
+            self,
+            partition_names,
+            metastore_conn_id='metastore_default',
+            poke_interval=60*3,
+            *args,
+            **kwargs):
+        super(NamedHivePartitionSensor, self).__init__(
+            poke_interval=poke_interval, *args, **kwargs)
+
+        if isinstance(partition_names, basestring):
+            raise TypeError('partition_names must be an array of strings')
+
+        self.metastore_conn_id = metastore_conn_id
+        self.partition_names = partition_names
+
+    def poke(self, context):
+
+        if not hasattr(self, 'hook'):
+            self.hook = hooks.HiveMetastoreHook(
+                metastore_conn_id=self.metastore_conn_id)
+
+        def poke_partition(partition):
+            schema, table_partition = partition.split('.')
+            table, partition = table_partition.split('/', 1)
+
+            logging.info(
+                'Poking for {schema}.{table}/{partition}'.format(**locals())
+            )
+            return self.hook.check_for_named_partition(
+                schema, table, partition)
+
+        return all(poke_partition(p) for p in self.partition_names)
```

Comment:
We should keep track of which partitions have been tested to exist. Otherwise, the partitions in the beginning of the list will be queried repeatedly.